src/commands/mail/address/update.tsx

```diff
@@ -0,0 +1,267 @@
+import { Flags } from "@oclif/core";
+import { assertStatus } from "@mittwald/api-client-commons";
+import { projectFlags } from "../../../lib/project/flags.js";
+import { ExecRenderBaseCommand } from "../../../rendering/react/ExecRenderBaseCommand.js";
+import {
+  makeProcessRenderer,
+  processFlags,
+} from "../../../rendering/process/process_flags.js";
+import { Text } from "ink";
+import { Success } from "../../../rendering/react/components/Success.js";
+import { ReactNode } from "react";
+import { ProcessRenderer } from "../../../rendering/process/process.js";
+import { mailAddressArgs, withMailAddressId } from "../../../lib/mail/flags.js";
+import crypto from "crypto";
+
+type UpdateResult = {
+  generatedPassword: string | null;
+};
+
+export default class Update extends ExecRenderBaseCommand<
+  typeof Update,
+  UpdateResult
+> {
+  static summary = "Update a mail address";
+  static description = `\
+    This command can be used to update a mail address in a project.
+    
+    A mail address is either associated with a mailbox, or forwards to another address.
+    
+    To set forwarding addresses, use the --forward-to flag.
+    
+    Use the --catch-all flag to make the mailbox a catch-all mailbox.
+  
+    When running this command with --generated-password the output will be the newly generated and set password.`;
+  static args = { ...mailAddressArgs };
+  static flags = {
+    ...projectFlags,
+    ...processFlags,
+    address: Flags.string({
+      char: "a",
+      summary: "mail address",
+    }),
+    "catch-all": Flags.boolean({
+      description: "make this a catch-all mail address",
+    }),
+    quota: Flags.integer({
+      description: "mailbox quota in mebibytes",
+    }),
+    password: Flags.string({
+      summary: "mailbox password",
+      description:
+        "This is the password that should be used for the mailbox; if omitted, the command will prompt interactively for a password.\n\nCAUTION: providing this flag may log your password in your shell history!",
+    }),
+    "random-password": Flags.boolean({
+      summary: "generate a random password",
+      description:
+        "This flag will cause the command to generate a random 32-character password for the mailbox; when running with --quiet, the address ID and the password will be printed to stdout, separated by a tab character.",
+    }),
+    "forward-to": Flags.string({
+      summary: "forward mail to another address",
+      multiple: true,
+      description:
+        "This flag will cause the mailbox to forward all incoming mail to the given address.\n\nNote: This flag is exclusive with --catch-all, --enable-spam-protection, --quota, --password and --random-password.",
+      relationships: [
+        {
+          type: "none",
+          flags: [
+            "catch-all",
+            "enable-spam-protection",
+            "quota",
+            "password",
+            "random-password",
+          ],
+        },
+      ],
+    }),
```

Comment:
Also, `create` and `update` share most of their flags. Perhaps it would make sense to extract these into a common `mailaddressFlags` object that can be spreaded into the flag definitions of both commands (see `mysql` commands, for exampe).

Reply:
I extracted `address` & `"catch-all"`. There are other shared flags, but they differ slightly between update and create.